src/tests/test_package_loading.py

```diff
@@ -752,7 +752,134 @@ def test_custom_lockfile_different_dir(selenium_standalone_noload, tmp_path):
             f"""
             let pyodide = await loadPyodide({{fullStdLib: false, lockFileURL: {lockfile_url!r} }});
             await pyodide.loadPackage("dummy_pkg", {{ checkIntegrity: false }});
-            return pyodide.runPython("import dummy_pkg;")
+            return pyodide.runPython("import dummy_pkg")
+            """
+        )
+
+
+def test_lock_file_contents_error(selenium_standalone_noload):
+    selenium = selenium_standalone_noload
+    message = "Error: Can't pass both lockFileContents and lockFileURL"
+    with pytest.raises(selenium.JavascriptException, match=message):
+        selenium.run_js(
+            """
+            await loadPyodide({
+                lockFileContents: "x",
+                lockFileURL: "y"
+            });
+            """
+        )
+
+
+def test_lock_file_contents_relative_file_name(selenium_standalone_noload, tmp_path):
+    selenium = selenium_standalone_noload
+    orig_lockfile = DIST_PATH / "pyodide-lock.json"
+    test_file_name = "dummy_pkg-0.1.0-py3-none-any.whl"
+    lockfile_content = json.loads(orig_lockfile.read_text())
+    lockfile_content["packages"] = {
+        "dummy-pkg": {
+            "name": "dummy_pkg",
+            "version": "0.1.0",
+            "unvendor_tests": False,
+            "sha256": "22fc6330153be71220aea157ab135c53c7d34ff1a6d1d1a4705c95eef1a6f262",
+            "depends": [],
+            "file_name": test_file_name,
+            "install_dir": "site",
+            "package_type": "package",
+            "imports": [],
+        }
+    }
+    message = 'Lock file file_name for package "dummy_pkg" is relative path "dummy_pkg-0.1.0-py3-none-any.whl" but no packageBaseUrl provided'
+    content = json.dumps(lockfile_content)
+    selenium.run_js(
+        """
+        const py = await loadPyodide({
+            lockFileContents: %s,
+        });
+        await py.loadPackage("dummy_pkg");
+        """
+        % content
+    )
+    assert message in selenium.logs
+
+
+def test_lockfilecontents_package_base_url(selenium_standalone_noload, tmp_path):
+    selenium = selenium_standalone_noload
+    orig_lockfile = DIST_PATH / "pyodide-lock.json"
+    test_file_name = "dummy_pkg-0.1.0-py3-none-any.whl"
+    test_file_path = Path(__file__).parent / "wheels" / test_file_name
+
+    lockfile_content = json.loads(orig_lockfile.read_text())
+    lockfile_content["packages"] = {
+        "dummy-pkg": {
+            "name": "dummy_pkg",
+            "version": "0.1.0",
+            "unvendor_tests": False,
+            "sha256": "22fc6330153be71220aea157ab135c53c7d34ff1a6d1d1a4705c95eef1a6f262",
+            "depends": [],
+            "file_name": test_file_name,
+            "install_dir": "site",
+            "package_type": "package",
+            "imports": [],
+        }
+    }
+    lockfile_content_json = json.dumps(lockfile_content)
+
+    shutil.copy(test_file_path, tmp_path / test_file_name)
+
+    with spawn_web_server(tmp_path) as web_server:
```

Comment:
How about using [httpserver](https://github.com/pyodide/pyodide/blob/654ad14f91841c48653f5928c9fe98c08cec0c9a/src/tests/test_package_loading.py#L935-L941) fixture instead? I would like to avoid using APIs like `spawn_web_server` outside of pytest-pyodide, which makes it difficult for us to modify pytest-pyodide.

Reply:
It's used in the next few tests, so if you want to get rid of it could you do it to the whole file in a separate PR?